service/raft/raft_address_map.hh

```diff
@@ -87,22 +87,22 @@ class raft_address_map {
         const raft::server_id& _entry_id;
     };
 
-    // An `inet_address` optionally equipped with a pointer to an entry
-    // in LRU list of 'expiring entries'. If the pointer is set, it means that this
-    // `timestamped_entry` is expiring; the corresponding LRU list entry contains
-    // the last access time and we periodically delete elements from the LRU list
-    // when they become too old.
+    // An `inet_address` optionally equipped with an optional field
+    // which is a member of a LRU list of 'expiring entries'.  If the field is set,
+    // it means that this `timestamped_entry` is expiring; the LRU list entry
+    // contains the last access time and we periodically delete elements from
+    // the LRU list when they become too old.
     struct timestamped_entry {
         gms::inet_address _addr;
-        std::unique_ptr<expiring_entry_ptr> _lru_entry;
+        std::optional<expiring_entry_ptr> _lru_entry;
```

Comment:
This makes non expiring `timestamped_entry` larger. We will mostly have non expiring entries. Not sure if it is important.

Reply:
Fine point. I'll leave the `unique_ptr` then.